### PR TITLE
[DCJ-616] Update unit and connected tests in GitHub Action to latest standards

### PR DESCRIPTION
**Pull request by fboulnois**
## Addresses

https://broadworkbench.atlassian.net/browse/DCJ-616

## Summary of changes

Removes use of `datarepo-actions` in unit and connected tests and prepares integration tests for further refactoring. The integration improvements will be done in a separate PR.

## Testing Strategy

Tested in branch, can also be tested manually with `workflow_dispatch`

Latest run (all green): https://github.com/DataBiosphere/jade-data-repo/actions/runs/10527581150

### CODE REVIEW

File: .github/workflows/int-and-connected-test-run.yml [hunk 1]
```diff
@@ -1,103 +1,42 @@
...
-    build/reports
-    build/spotless
-    build/test-results
-    build/jacocoHtml
 on:
-  pull_request:
-    branches:
-      - develop
-    paths:
-      - '!*'
```
Comment (fboulnois) [inline .github/workflows/int-and-connected-test-run.yml:30]: Reviewer note: I prefer simplifying which paths we run the action vs ignoring certain directories. I might consider excluding the `tools/` and `scripts/` directories from running this action.

File: .github/workflows/int-and-connected-test-run.yml [hunk 2]
```diff
@@ -1,103 +1,42 @@
 name: Unit, Smoke, Connected and Integration tests
-env:
-  K8_CLUSTER: 'integration-master'
```
Comment (fboulnois) [inline .github/workflows/int-and-connected-test-run.yml:3]: Reviewer note: I think it's better to pass the minimum number of required environment variables to each step to see *exactly* what they need to run

File: .github/workflows/int-and-connected-test-run.yml [hunk 3]
```diff
@@ -1,103 +1,42 @@
...
-      job-status: ${{ job.status }}
+  test_unit:
+    name: Run unit tests
+    runs-on: ubuntu-latest
     timeout-minutes: 60
-    strategy:
-      matrix:
-        os: [ubuntu-latest]
-    if: "!contains( github.event.pull_request.labels.*.name, 'skip-ci')"
-    runs-on: ${{ matrix.os }}
```
Comment (fboulnois) [inline .github/workflows/int-and-connected-test-run.yml:57]: Reviewer note: we've never really used `skip-ci`, and our strategy matrix is just to use `ubuntu-latest`, so I simplified this as well

File: .github/workflows/int-and-connected-test-run.yml [hunk 4]
```diff
@@ -1,103 +1,42 @@
...
-    services:
-      postgres:
-        image: postgres:11
-        env:
-          POSTGRES_USER: postgres
-          POSTGRES_PASSWORD: postgres
-          POSTGRES_DB: postgres
-        ports:
-          - 5432:5432
-        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
```
Comment (fboulnois) [inline .github/workflows/int-and-connected-test-run.yml:69]: Reviewer note: we don't need Postgres for unit tests. Makes sense!

File: .github/workflows/int-and-connected-test-run.yml [hunk 5]
```diff
@@ -1,103 +1,42 @@
...
-  # The default Azure credentials to use to authenticate
-  AZURE_CREDENTIALS_APPLICATIONID: 0e29ec36-04e8-44d5-ae7c-50dc15135571
-  AZURE_CREDENTIALS_HOMETENANTID: fad90753-2022-4456-9b0a-c7e5b934e408
-  JADE_USER_EMAIL: connected-tdr-user@notarealemail.org
-  RBS_CLIENT_CREDENTIAL_FILE_PATH: rbs-tools-sa.json
-  AZURE_CREDENTIALS_SECRET: ${{ secrets.AZURE_CREDENTIALS_SECRET }}
-  AZURE_SYNAPSE_SQLADMINUSER: ${{ secrets.AZURE_SYNAPSE_SQLADMINUSER }}
-  AZURE_SYNAPSE_SQLADMINPASSWORD: ${{ secrets.AZURE_SYNAPSE_SQLADMINPASSWORD }}
-  AZURE_SYNAPSE_WORKSPACENAME: tdr-snps-int-east-us-ondemand.sql.azuresynapse.net
-  CACHE_PATHS: |
```
Comment (fboulnois) [inline .github/workflows/int-and-connected-test-run.yml:19]: Reviewer note: we use the automatic caching facilities of the `actions/setup-java@v4` action, which is much more robust, and we skip caching build reports

File: .github/workflows/int-and-connected-test-run.yml [hunk 6]
```diff
@@ -1,103 +1,42 @@
...
+        env:
+          GOOGLE_APPLICATION_CREDENTIALS: 'jade-dev-account.json'
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          TDR_LOG_APPENDER: 'Console-Standard'
+        run: |
+          # extract service account credentials
+          base64 --decode <<< ${{ secrets.SA_B64_CREDENTIALS }} > ${GOOGLE_APPLICATION_CREDENTIALS}
+          # assemble code, run unit tests, and generate scan
+          ./gradlew assemble
+          ./gradlew check --scan jacocoTestReport sonar
```
Comment (fboulnois) [inline .github/workflows/int-and-connected-test-run.yml:37]: Here is the minimal amount of code required to run unit tests in a hermetic environment. No more `datarepo-actions`!

File: .github/workflows/int-and-connected-test-run.yml (hunk 4 above)
Comment (okotsopoulos) [reply .github/workflows/int-and-connected-test-run.yml:69]: Oh I'm a bit surprised by this!  Our unit tests contain a number of embedded database tests, do those need a Postgres?

Comment (fboulnois) [reply .github/workflows/int-and-connected-test-run.yml:69]: They create their own Postgres instances using `testcontainers`.

Comment (okotsopoulos) [reply .github/workflows/int-and-connected-test-run.yml:69]: Rad!  That's great.

File: .github/workflows/int-and-connected-test-run.yml [hunk 7]
```diff
@@ -1,153 +1,85 @@
...
-          path: build/reports
-          retention-days: 5
-  deploy_test_integration:
-    name: "Run integration and smoke tests"
-    outputs:
-      job-status: ${{ job.status }}
-      api_image_tag: ${{ steps.configuration.outputs.git_hash }}
+          # assemble code and run connected tests
+          ./gradlew assemble
+          ./gradlew testConnected --scan
```
Comment (fboulnois) [inline .github/workflows/int-and-connected-test-run.yml:78]: Reviewer note: the connected tests *also* do not need a dedicated Postgres deployment.

Comment (fboulnois) [reply .github/workflows/int-and-connected-test-run.yml:78]: https://github.com/DataBiosphere/jade-data-repo/actions/runs/10477254542

File: .github/workflows/int-and-connected-test-run.yml [hunk 8]
```diff
@@ -159,137 +91,76 @@ jobs:
...
           DESIRED_GITHASH: ${{ steps.configuration.outputs.git_hash }}
           DEPLOYMENT_TYPE: 'api'
-      - name: "Run Test Runner smoke tests via Gradle"
+      - name: Run test runner smoke tests via Gradle
         uses: broadinstitute/datarepo-actions/actions/main@0.74.0
         with:
           actions_subcommand: 'gradletestrunnersmoketest'
           sa_b64_credentials: ${{ secrets.SA_B64_CREDENTIALS }}
-      - name: "Run integration tests via Gradle"
+      - name: Run integration tests via Gradle
```
Comment (fboulnois) [inline .github/workflows/int-and-connected-test-run.yml:162]: Reviewer note: per discussion with DE and TDR team this morning, the integration tests will be handled in a separate PR.

File: .github/workflows/int-and-connected-test-run.yml [hunk 9]
```diff
@@ -159,137 +91,76 @@ jobs:
...
           actions_subcommand: 'k8_checknamespace_clean'
           sa_b64_credentials: ${{ secrets.SA_B64_CREDENTIALS }}
-      - name: "Clean whitelisted Runner IP"
+      - name: Clean whitelisted Runner IP
         if: always()
         uses: broadinstitute/datarepo-actions/actions/main@0.74.0
         with:
           actions_subcommand: 'gcp_whitelist_clean'
           sa_b64_credentials: ${{ secrets.SA_B64_CREDENTIALS }}
-  report-to-sherlock:
```
Comment (fboulnois) [inline .github/workflows/int-and-connected-test-run.yml:234]: Reviewer note: do we still want to report any of this information?

File: .github/workflows/int-and-connected-test-run.yml [hunk 10]
```diff
@@ -158,138 +90,98 @@ jobs:
         ports:
           - 5432:5432
         options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+    env:
```
Comment (fboulnois) [inline .github/workflows/int-and-connected-test-run.yml:95]: Note for reviewers: these are required here because otherwise the shell scripts will fail. These will all go away once integration tests are refactored.

**Comment by fboulnois** [comment]
Latest run (successful): https://github.com/DataBiosphere/jade-data-repo/actions/runs/10527581150

File: .github/workflows/int-and-connected-test-run.yml [hunk 11]
```diff
@@ -158,138 +90,98 @@ jobs:
...
     steps:
-      - name: "Checkout code"
-        uses: actions/checkout@v3
-      - name: "Cache build"
-        uses: actions/cache@v3
-        with:
-          path: ${{ env.CACHE_PATHS }}
-          key: ${{ runner.os }}-build-integration
-      - name: "Whitelist Runner IP"
+      - name: Checkout code
```
Comment (fboulnois) [inline .github/workflows/int-and-connected-test-run.yml:103]: Note for reviewers: these changes were not strictly necessary, but will make the upcoming integration changes easier to implement.

File: .github/workflows/int-and-connected-test-run.yml [hunk 12]
```diff
@@ -158,138 +90,98 @@ jobs:
...
           echo "Latest git hash for this branch: $git_hash"
-      - name: "Wait for deployment to come back online"
+      - name: Wait for deployment to come back online
         uses: broadinstitute/datarepo-actions/actions/wait-for-deployment@0.74.0
         timeout-minutes: 20
         env:
           DESIRED_GITHASH: ${{ steps.configuration.outputs.git_hash }}
           DEPLOYMENT_TYPE: 'api'
-      - name: "Run Test Runner smoke tests via Gradle"
+      - name: Run test runner smoke tests via Gradle
```
Comment (fboulnois) [inline .github/workflows/int-and-connected-test-run.yml:152]: Note for reviewers: the smoke tests can be split out further in the refactoring of the integration piece, which will save time when these are running concurrently.

File: .github/workflows/int-and-connected-test-run.yml [hunk 13]
```diff
@@ -158,138 +90,98 @@ jobs:
...
         uses: broadinstitute/datarepo-actions/actions/main@0.74.0
+        env:
+          # required for azure tests
+          AZURE_CREDENTIALS_APPLICATIONID: 0e29ec36-04e8-44d5-ae7c-50dc15135571
+          AZURE_CREDENTIALS_HOMETENANTID: fad90753-2022-4456-9b0a-c7e5b934e408
+          AZURE_CREDENTIALS_SECRET: ${{ secrets.AZURE_CREDENTIALS_SECRET }}
+          # unset the GitHub Action default JAVA_HOME to run with JDK 17
+          JAVA_HOME:
+          # output plain logs instead of json
+          TDR_LOG_APPENDER: 'Console-Standard'
```
Comment (fboulnois) [inline .github/workflows/int-and-connected-test-run.yml:172]: Reviewer note: this is the minimum number of environment variables actually required to run integration tests.

File: .github/workflows/int-and-connected-test-run.yml [hunk 14]
```diff
@@ -158,138 +90,98 @@ jobs:
...
         timeout-minutes: 20
         env:
           DESIRED_GITHASH: ${{ steps.configuration.outputs.git_hash }}
           DEPLOYMENT_TYPE: 'api'
-      - name: "Run Test Runner smoke tests via Gradle"
+      - name: Run test runner smoke tests via Gradle
         uses: broadinstitute/datarepo-actions/actions/main@0.74.0
+        env:
+          # unset the GitHub Action default JAVA_HOME to run with JDK 17
+          JAVA_HOME:
```
Comment (fboulnois) [inline .github/workflows/int-and-connected-test-run.yml:156]: Reviewer note: there seems to be some sort of conflict between the version of Java that the shell scripts expect and the version inside the GitHub Action, but this should go away once the actions are fully transitioned away from shell scripts.

File: .github/workflows/int-and-connected-test-run.yml [hunk 15]
```diff
@@ -1,153 +1,85 @@
...
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+          cache: 'gradle'
+      - name: Run connected tests
+        env:
+          # required for azure tests
+          AZURE_CREDENTIALS_APPLICATIONID: 0e29ec36-04e8-44d5-ae7c-50dc15135571
+          AZURE_CREDENTIALS_HOMETENANTID: fad90753-2022-4456-9b0a-c7e5b934e408
```
Comment (rushtong) [inline .github/workflows/int-and-connected-test-run.yml:55]: Is that the case here that these are similar to google client id values that are not intended to be secret?

File: .github/workflows/int-and-connected-test-run.yml [hunk 16]
```diff
@@ -1,153 +1,85 @@
...
-      - 'src/**'
-      - 'gradle**'
-      - 'gradle/**'
-      - '**.gradle'
-      - 'Dockerfile'
-      - 'datarepo-clienttests/**'
-      - '.github/workflows/int-and-connected-test-run.yml'
-      - '.swagger-codegen-ignore'
   workflow_dispatch: {}
+  pull_request:
```
Comment (rushtong) [inline .github/workflows/int-and-connected-test-run.yml:4]: What does an empty `pull_request` block do? I can't find any docs specifically on this.

File: .github/workflows/int-and-connected-test-run.yml (hunk 1 above)
Comment (rushtong) [reply .github/workflows/int-and-connected-test-run.yml:30]: I have mixed thoughts on this. It makes sense that we wouldn't want to run this in certain cases, but if the action is marked as required, it puts PRs against those cases in limbo requiring admin override. For that reason, I favor ~leaving this out.~ the approach this PR makes.

**Review by rushtong — APPROVED** [review]
Amazing work, thank you for your persistence! 👍🏽

File: .github/workflows/int-and-connected-test-run.yml (hunk 16 above)
Comment (fboulnois) [reply .github/workflows/int-and-connected-test-run.yml:4]: It means it will run when a `pull_request` is opened, synchronized, or reopened: https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request

File: .github/workflows/int-and-connected-test-run.yml (hunk 15 above)
Comment (fboulnois) [reply .github/workflows/int-and-connected-test-run.yml:55]: Yes, that's correct.

File: .github/workflows/int-and-connected-test-run.yml (hunk 1 above)
Comment (pshapiro4broad) [reply .github/workflows/int-and-connected-test-run.yml:30]: Why change this now? If the idea is to convert and preserve behavior, I would move this change to a different PR so its impact can be evaluated separately.

Are there known problems with the current list of paths that changing it would fix? (E.g., a case where tests aren't being run but they should be?)

Also, if we do work on this part of the workflow action, it would be nice to address the issue where PRs that skip tests run require a force override when merging due to "pending" test results.

File: .github/workflows/int-and-connected-test-run.yml (hunk 4 above)
Comment (pshapiro4broad) [reply .github/workflows/int-and-connected-test-run.yml:69]: > Makes sense

It does today but before the embedded db support was added, all Dao unit tests required postgres. In many (most?) other terra service projects, Dao unit tests depend on postgres, so that behavior isn't uncommon. The use of testcontainers is more uncommon IMO.

File: .github/workflows/int-and-connected-test-run.yml (hunk 6 above)
Comment (pshapiro4broad) [reply .github/workflows/int-and-connected-test-run.yml:37]: Is there a reason to not add `assemble` to this list? I think it would be slightly more efficient. Also I would put `--scan` first so it's clearer that it's its own option, not a combined option list `--scan jacocoTestReport`
```
          ./gradlew --scan assemble check jacocoTestReport sonar
```

For a future PR, it would be nice to combine these gradle tasks into a new gradle task so it could be run with one argument.

File: .github/workflows/int-and-connected-test-run.yml [hunk 17]
```diff
@@ -1,153 +1,85 @@
...
-          name: junit-test-reports-for-connected
-          path: build/reports
-          retention-days: 5
-  deploy_test_integration:
-    name: "Run integration and smoke tests"
-    outputs:
-      job-status: ${{ job.status }}
-      api_image_tag: ${{ steps.configuration.outputs.git_hash }}
+          # assemble code and run connected tests
+          ./gradlew assemble
```
Comment (pshapiro4broad) [inline .github/workflows/int-and-connected-test-run.yml:77]: Did the old workflows not use a build cache for gradle (to share build outputs between jobs)? Something to think about for a future PR; I have't used https://github.com/gradle/actions/blob/main/docs/setup-gradle.md but it looks interesting.

Comment (pshapiro4broad) [inline .github/workflows/int-and-connected-test-run.yml:77]: Both this and the previous test run use `assemble` before running the tests. Why is this task run? It looks like it invokes the same tasks as the test task, except for: resolveMainClassName, bootJar, jar, assemble.

Maybe using `assemble` is from a workflow that needed a `.jar` build for some reason?

File: .github/workflows/int-and-connected-test-run.yml [hunk 18]
```diff
@@ -158,138 +90,98 @@ jobs:
...
-      - name: "Deploy to cluster with Helm"
+      - name: Deploy to cluster with Helm
         uses: broadinstitute/datarepo-actions/actions/main@0.74.0
         with:
           actions_subcommand: 'helmdeploy'
           helm_create_secret_manager_secret_version: 0.0.8
-          helm_datarepo_api_chart_version: 0.0.739
-          helm_datarepo_ui_chart_version: 0.0.360
+          helm_datarepo_api_chart_version: 0.0.744
+          helm_datarepo_ui_chart_version: 0.0.362
```
Comment (pshapiro4broad) [inline .github/workflows/int-and-connected-test-run.yml]: I'm unfamiliar with how this works. How are these versions updated normally?

File: .github/workflows/int-and-connected-test-run.yml (hunk 14 above)
Comment (pshapiro4broad) [reply .github/workflows/int-and-connected-test-run.yml:156]: That's odd, it'll be good to get rid of this when that's done

File: .github/workflows/int-and-connected-test-run.yml (hunk 9 above)
Comment (pshapiro4broad) [reply .github/workflows/int-and-connected-test-run.yml:234]: How would sherlock know about a branch version if this was removed? Is there something else that's registering branch builds with sherlock? If anything, it'd be nice to enable publishing to sherlock without `needs: deploy_test_integration` as it'd b nice to use a branch build in a bee without having to wait for integration tests to pass first.

**Review by pshapiro4broad — APPROVED** [review]
Looks ok overall, I don't think we want to remove the sherlock registration job though.

File: .github/workflows/int-and-connected-test-run.yml (hunk 6 above)
Comment (fboulnois) [reply .github/workflows/int-and-connected-test-run.yml:37]: I did this to mimic the current behavior: https://github.com/broadinstitute/datarepo-actions/blob/master/actions/main/src/gradleinttest.sh#L61

File: .github/workflows/int-and-connected-test-run.yml (hunk 17 above)
Comment (fboulnois) [reply .github/workflows/int-and-connected-test-run.yml:77]: The old workflows use the `cache` action, instead of the one built into `actions/setup-java@v4`. The java one is more robust, supports gradle natively, and doesn't require a cache key.

Comment (fboulnois) [reply .github/workflows/int-and-connected-test-run.yml:77]: This is to mimic the current behavior: https://github.com/broadinstitute/datarepo-actions/blob/master/actions/main/src/gradleinttest.sh#L61

File: .github/workflows/int-and-connected-test-run.yml (hunk 18 above)
Comment (fboulnois) [reply .github/workflows/int-and-connected-test-run.yml]: There's a separate action that bumps these and keeps them synced across environments. It's another wart that will be removed once we transition away from the current integration testing approach.

File: .github/workflows/int-and-connected-test-run.yml (hunk 9 above)
Comment (fboulnois) [reply .github/workflows/int-and-connected-test-run.yml:234]: It's not clear what the `sherlock` functionality does. Is this something we need?

File: .github/workflows/int-and-connected-test-run.yml (hunk 1 above)
Comment (fboulnois) [reply .github/workflows/int-and-connected-test-run.yml:30]: One issue is that this approach doesn't follow GitHub best practices. Right now we're effectively telling GitHub Actions "ignore all paths/files except for these paths/files" (effectively an allowlist). I would prefer having a denylist using `paths-ignore` instead, which is the suggested mechanism: https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore

I think the current approach may also be related to why we're having the bug where we need to force override.

File: .github/workflows/int-and-connected-test-run.yml (hunk 18 above)
Comment (fboulnois) [reply .github/workflows/int-and-connected-test-run.yml]: (note that I didn't manually update these values, these came from restoring the integration test piece)

File: .github/workflows/int-and-connected-test-run.yml (hunk 4 above)
Comment (rushtong) [reply .github/workflows/int-and-connected-test-run.yml:69]: For a point of reference, we migrated Consent to postgres testcontainer-based unit tests in March 2020.

File: .github/workflows/int-and-connected-test-run.yml (hunk 1 above)
Comment (rushtong) [reply .github/workflows/int-and-connected-test-run.yml:30]: > I think the current approach may also be related to why we're having the bug where we need to force override.

This is correct. I've run into this once where it was unclear why tests were not being run even though they are marked as required in the branch protection rules. It required a repo admin to override merge the PR.

File: .github/workflows/int-and-connected-test-run.yml (hunk 9 above)
Comment (fboulnois) [reply .github/workflows/int-and-connected-test-run.yml:234]: I restored this functionality, but I'm not sure it's necessary.

Comment (okotsopoulos) [reply .github/workflows/int-and-connected-test-run.yml:234]: We do want to report versions to Sherlock: doing so is what makes Beehive aware of our feature branches, so that devs can deploy them to their Bees.  If a comment or additional description would help make that clear (so that this step isn't inadvertently removed in the future), I'm all for adding them.

For reference, here is the current comment -- maybe what would help is a slight elaboration to make it clear why we want to report the app version.

>     # Always attempt to run if pull_request, as we want to report the appVersion even if the tests fail.

--

> If anything, it'd be nice to enable publishing to sherlock without needs: deploy_test_integration as it'd b nice to use a branch build in a bee without having to wait for integration tests to pass first.

@pshapiro4broad I agree with you, that would be nice. I'm not sure it's as simple as just removing the `needs: deploy_test_integration` so leaving it in for now is my preference.  See related ticket: https://broadworkbench.atlassian.net/browse/DCJ-38

File: .github/workflows/int-and-connected-test-run.yml (hunk 2 above)
Comment (okotsopoulos) [reply .github/workflows/int-and-connected-test-run.yml:3]: I agree with this shift as a general rule.  What would you think about making an exception for this variable?  It feels like a general setting that we want for all tests.

```
  # This will make test logs not use JSON
  TDR_LOG_APPENDER: 'Console-Standard'
```

File: .github/workflows/int-and-connected-test-run.yml (hunk 4 above)
Comment (okotsopoulos) [reply .github/workflows/int-and-connected-test-run.yml:69]: This is interesting -- it makes more sense to me that we'd have this Postgres dependency left lying around from our pre-embedded DB test days.

File: .github/workflows/int-and-connected-test-run.yml (hunk 6 above)
Comment (okotsopoulos) [reply .github/workflows/int-and-connected-test-run.yml:37]: I like @pshapiro4broad 's suggestions here, and agree with making any behavior changes here in a future PR.

File: .github/workflows/int-and-connected-test-run.yml (hunk 17 above)
Comment (okotsopoulos) [reply .github/workflows/int-and-connected-test-run.yml:77]: I'd be interested to see if this is really necessary, but for any future work rather than in this PR.

File: .github/workflows/int-and-connected-test-run.yml (hunk 12 above)
Comment (okotsopoulos) [reply .github/workflows/int-and-connected-test-run.yml:152]: In this case, "Test Runner" refers to a Data Biosphere library called Test Runner, so I think the original capitalization in the name makes sense.

https://github.com/DataBiosphere/terra-test-runner

File: .github/workflows/int-and-connected-test-run.yml (hunk 14 above)
Comment (okotsopoulos) [reply .github/workflows/int-and-connected-test-run.yml:156]: Agreed, very strange.

**Review by okotsopoulos — APPROVED** [review]
Fantastic efforts here -- I found it easiest to review and appreciate when looking at the latest version outside of diff view, in case that helps others.

I left a few questions, comments, and additional info about Sherlock reporting but nothing too big.

File: .github/workflows/int-and-connected-test-run.yml (hunk 6 above)
Comment (fboulnois) [reply .github/workflows/int-and-connected-test-run.yml:37]: I'll roll in these changes in the integration PR since the feedback cycle running connected tests and integration tests is very slow.

File: .github/workflows/int-and-connected-test-run.yml (hunk 1 above)
Comment (pshapiro4broad) [reply .github/workflows/int-and-connected-test-run.yml:30]: > I think the current approach may also be related to why we're having the bug where we need to force override.

The "bug" is in how this repo is configured. Changing this workflow to use `paths` vs `paths-ignore` shouldn't make a difference. The problem is that the repo says "Job X is required" but include/exclude config prevents the job X from running, so its status will never be reported. Although maybe having this config directly on the job, vs on a subworkflow, will change how this is reported.

>  I would prefer having a denylist using paths-ignore instead, which is the suggested mechanism

I don't see how the GitHub docs recommend `paths` vs `paths-ignore`; from my reading it describes both without specifying one as preferred over the other. It does say that `paths` is the most flexible as it allows both include and exclude path regex, where as `paths-ignore` doesn't support `!`.
